src/app/AttributePathExpandIterator.cpp

```diff
@@ -65,7 +65,8 @@ AttributePathExpandIterator::AttributePathExpandIterator(ObjectList<AttributePat
                   "Our index won't be able to hold the value we need to hold.");
     static_assert(std::is_same<decltype(mGlobalAttributeIndex), uint8_t>::value,
                   "If this changes audit all uses where we set to UINT8_MAX");
-    mGlobalAttributeIndex = UINT8_MAX;
+    mGlobalAttributeIndex    = UINT8_MAX;
+    mGlobalAttributeEndIndex = 0;
```

Comment:
I am not sure this is the right fix.  What I think would make more sense is in PrepareAttributeIndexRange to change the `if (mAttributeIndex == UINT16_MAX)` bit to have an `else` block that sets `mGlobalAttributeEndIndex` to 0 and `mGlobalAttributeIndex` to UINT8_MAX.

That way PrepareAttributeIndexRange always initializes all the members involved properly (which right now I agree it does not if we are in the non-wildcard case and mAttributeIndex ends up non-max.

That is, the change above makes the value _initialized_, but it's not obviously clear that it ends up with the _right_ value later.

Are you willing to test whether my proposed change above addresses the valgrind issue too, @DamMicSzm ?